src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BackgroundAspectEffect/BackgroundAspectEffectRouter.ios.cs

```diff
@@ -0,0 +1,152 @@
+
+		protected override void OnDetached()
+			=> ClearBackgroundImage();
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (!args.PropertyName.Equals(BackgroundAspectEffect.AspectProperty.PropertyName))
+				return;
+
+			cancellationTokenSource.Cancel();
+
+			Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
+		}
+
+		async Task ApplyBackgroundImageAsync()
+		{
+			var contentPage = Element as ContentPage;
+
+			if (Element is null || contentPage is null || contentPage.BackgroundImageSource is null)
+				return;
+
+			var image = await GetBackgroundImageFromSource(contentPage.BackgroundImageSource);
+
+			if (image is null)
+				return;
+
+			var fallbackColor = contentPage.BackgroundColor.ToUIColor();
+
+			Device.BeginInvokeOnMainThread(() => ApplyImageToView(image, fallbackColor));
+		}
+
+		void ApplyImageToView(UIImage image, UIColor color)
+		{
+			var effectParam = BackgroundAspectEffect.GetAspect(Element);
+
+			imageView = CreateImageView(image, effectParam);
+
+			Container.BackgroundColor = color;
+
+			Container.InsertSubview(imageView, 0);
+
+			ApplyConstraints(imageView, Container);
+		}
+
+		void ClearBackgroundImage()
+		{
+			if (imageView != null)
+			{
+				imageView.RemoveFromSuperview();
+
+				imageView.Dispose();
+			}
+
+			// TODO: Is this enough to cleanup?
+			cancellationTokenSource.Cancel();
+			cancellationTokenSource.Dispose();
+		}
+
+		async Task<UIImage> GetBackgroundImageFromSource(ImageSource imageSource)
+		{
+			IImageSourceHandler handler;
+
+			if (imageSource is FileImageSource)
+			{
+				handler = new FileImageSourceHandler();
+			}
+			else if (imageSource is StreamImageSource)
+			{
+				handler = new StreamImagesourceHandler();
+			}
+			else if (imageSource is UriImageSource)
+			{
+				handler = new ImageLoaderSourceHandler();
+			}
+			else
+			{
+				throw new NotSupportedException($"ImageSource is not supported by this effect: {imageSource.GetType().Name}");
+			}
+
+			return await handler.LoadImageAsync(imageSource);
+		}
+
+		UIImageView CreateImageView(UIImage uIImage, Aspect aspect)
+		{
+			var imageView = new UIImageView();
+			imageView.Image = uIImage;
+			imageView.ContentMode = GetContentModeForAspect(aspect);
+			imageView.TranslatesAutoresizingMaskIntoConstraints = false;
+
+			return imageView;
+		}
+
+		void ApplyConstraints(UIView child, UIView parent)
+		{
+			var constraints = new NSLayoutConstraint[]
+			{
+				child.TopAnchor.ConstraintEqualTo(parent.TopAnchor),
+				child.LeadingAnchor.ConstraintEqualTo(parent.LeadingAnchor),
+				child.TrailingAnchor.ConstraintEqualTo(parent.TrailingAnchor),
+				child.BottomAnchor.ConstraintEqualTo(parent.BottomAnchor)
+			};
+
+			NSLayoutConstraint.ActivateConstraints(constraints);
+		}
+
+		UIViewContentMode GetContentModeForAspect(Aspect aspect)
+		{
+			switch (aspect)
```

Comment:
if you can use the new switch expression will be great. I believe that IDE can do it for you (;

Reply:
I'll give it a go!